lib/reek/detector_repository.rb

```diff
@@ -35,31 +35,25 @@ def initialize(smell_types: self.class.smell_types,
                    configuration: {})
       @configuration = configuration
       @smell_types   = smell_types
-      @detectors     = smell_types.map { |klass| klass.new configuration_for(klass) }
     end
 
     def examine(context)
-      smell_detectors_for(context.type).flat_map do |detector|
-        detector.run_for(context)
+      smell_detectors_for(context.type).flat_map do |klass|
+        detector = klass.new configuration: configuration_for(klass), context: context
+        detector.run
       end
     end
 
     private
 
-    attr_reader :configuration, :smell_types, :detectors
+    attr_reader :configuration, :smell_types
 
     def configuration_for(klass)
       configuration.fetch klass, {}
     end
 
     def smell_detectors_for(type)
-      enabled_detectors.select do |detector|
-        detector.contexts.include? type
-      end
-    end
-
-    def enabled_detectors
-      detectors.select { |detector| detector.config.enabled? }
+      smell_types.select { |detector| detector.contexts.include? type }
```

Comment:
There's a cop (which is currently disabled by default) `Style/MethodCallWithArgsParentheses` which checks for the use of parens around method calls with arguments.

It is pretty reasonable if run with `IgnoreMacros: true` (default) and you can selectively whitelist methods as well. I am pushing to make it enabled in the next release of RuboCop, but if you're brave you can enable it now. 🙂